src/gitWorkflow.js

```diff
@@ -0,0 +1,197 @@
+'use strict'
+
+const path = require('path')
+const execa = require('execa')
+const fsp = require('fs-promise')
+const debug = require('debug')('lint-staged:git')
+
+let workingCopyTree = null
+let indexTree = null
+let formattedIndexTree = null
+
+function getAbsolutePath(dir) {
+  return path.isAbsolute(dir) ? dir : path.resolve(dir)
+}
+
+function getCmdArgs(gitDir) {
+  if (gitDir) {
+    return ['--git-dir', getAbsolutePath(gitDir)]
+  }
+  return []
+}
+
+async function execGit(cmd, options) {
+  const cwd = options && options.cwd ? options.cwd : process.cwd()
+  const gitDir = options && options.gitDir
+  debug('Running git command', cmd)
+  try {
+    const { stdout } = await execa('git', getCmdArgs(gitDir).concat(cmd), {
+      cwd: getAbsolutePath(cwd)
+    })
+    return stdout
+  } catch (err) {
+    throw new Error(err)
```

Comment:
Not sure what you mean. What wrapping? Without it will be unhandled promise case.

Reply:
What happens in the unhandled case?

I believe newer versions of v8 (therefore Node) handle unhandled promise rejections as regular errors now, if that's what you were trying to guard against. Maybe it's good to play it safe, just in case, for people on older versions of Node.